packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -739,6 +835,91 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
 
     return autoLinkAllowedTLDs.includes(tld);
   }
+
+  private findURL(input: string): string | undefined {
+    for (const match of findMatches(input, this.options.autoLinkRegex)) {
+      const urlMatch = getMatchString(match);
+
+      if (!this.isValidUrl(urlMatch)) {
+        continue;
+      }
+
+      return urlMatch;
+    }
+
+    return;
+  }
+
+  private getURLEndIndex({
+    url,
+    startIndex = 0,
+    text,
+  }: {
+    url?: string;
+    startIndex?: number;
+    text: string;
+  }) {
+    const newUrl = createNewURL(text);
+
+    // Check for trailing punctuation and adjacent characters
+    if (newUrl) {
+      const endsWithPunctuation = (input: string) =>
+        this.options.adjacentPunctuations.includes(input.slice(-1));
+
+      const adjacentCharCount = getTrailingCharCount({
+        input: text,
+        url,
+      });
+
+      const { pathname, search } = newUrl;
+
+      let decoded;
+
+      try {
+        decoded = decodeURI(pathname.slice(1) + search);
```

Comment:
I'm not sure I see the value in decoding a URL to check for special characters?

In my mind if they are URL encoded, by definition they are part of the URL.

Reply:
That is a good point

Reply:
Let me just remove all of that we don't need to use `URL()` just to get the path.